packages/server/src/api/controllers/row/utils/basic.ts

```diff
@@ -108,21 +126,30 @@ export function basicProcessing({
       thisRow[fieldName] = value
     }
   }
+
+  if (sdk.views.isView(source)) {
+    for (const key of Object.keys(helpers.views.calculationFields(source))) {
+      thisRow[key] = row[key]
+    }
+  }
+
   let columns: string[] = Object.keys(table.schema)
-  if (!sqs) {
-    thisRow._id = generateIdForRow(row, table, isLinked)
-    thisRow.tableId = table._id
-    thisRow._rev = "rev"
-    columns = columns.concat(PROTECTED_EXTERNAL_COLUMNS)
-  } else {
-    columns = columns.concat(PROTECTED_EXTERNAL_COLUMNS)
-    for (let internalColumn of [...PROTECTED_INTERNAL_COLUMNS, ...columns]) {
-      thisRow[internalColumn] = extractFieldValue({
-        row,
-        tableName: table._id!,
-        fieldName: internalColumn,
-        isLinked,
-      })
+  if (!isCalculationView) {
```

Comment:
Would it be possible to avoid the double `if` statement? I sometimes find these hard to reason about, its easier to read `if (!isCalculationView && !sqs) .... else if (!isCalculationView) ...` as each block is quite clear in its intentions.